src/query/service/src/interpreters/interpreter_factory.rs

```diff
@@ -53,17 +53,25 @@ pub struct InterpreterFactory;
 /// Such as: Plan::Query -> InterpreterSelectV2
 impl InterpreterFactory {
     #[async_backtrace::framed]
-    pub async fn get(ctx: Arc<QueryContext>, plan: &Plan) -> Result<InterpreterPtr> {
+    pub async fn get(
+        ctx: Arc<QueryContext>,
+        plan: &Plan,
+        is_explain_pipeline: bool,
```

Comment:
`is_explain_pipeline` could be derived from plan.

```
let is_explain_pipeline = match plan ...
```

Reply:
that's better, thanks

Reply:
we can't use this method, when we arrive here, it's the real plan like delete/update, not explain_pipeline plan.